### ISSUE TITLE
Issue with installing on Ubuntu

### ISSUE THREAD

**Issue by kq1231** [wassimj/topologicPy#24]
As-salamu 'alaikum! 
I have followed the steps of installing topologic. The problem is that I cannot install Blender using apt as it gives me some weird errors. I don't know why but I think it's because of some dependencies that are missing. I tried installing them but I failed. So isn't there a method where I can just take the zip and install it in Blender as an add-on like we usually do? I am able to install the modules for topologic but I get an error when I try installing the zip file (topologic-sverchok-main.zip): 
![Screenshot_20210809_215528](https://user-images.githubusercontent.com/88677687/128744264-7bec8ed5-0546-4200-83bf-44bc38733670.png)
Any ideas? 
And this is the problem I get when installing blender through apt: 
![Screenshot_20210809_215752](https://user-images.githubusercontent.com/88677687/128744609-68a3255e-ad3e-4c65-9377-d7e4526da44f.png)

**Comment by wassimj** [comment]
For Linux, I highly suggest you install homemaker by Bruno Postle. It includes Topologic. From what I see, some paths are wrong. It seems to be confusing Windows and Linux paths. I am working on a much simpler version to install, but it is delayed

**Comment by kq1231** [comment]
Okay. I'll try and let you know.

**Comment by kq1231** [comment]
![Screenshot_20210809_223154](https://user-images.githubusercontent.com/88677687/128748705-e9fe947a-e3ac-45ee-8691-d8a9c27ceedc.png)
Now it's giving me a problem when I'm installing homebrew.

**Comment by wassimj** [comment]
Doesn’t homemaker include instructions to install Topologic?

**Comment by kq1231** [comment]
I have installed topologicpy through my terminal as well but still to no avail

**Comment by kq1231** [comment]
> Doesn’t homemaker include instructions to install Topologic?

Yes it does and I did install it but it doesn't seem to work still. Is this an issue with snaps?

**Comment by wassimj** [comment]
I really don’t know much about snaps in Linux. Best to ask Bruno for help.

**Comment by kq1231** [comment]
Okay. Thanks for your help.

**Comment by kq1231** [comment]
I really appreciate it Wassim!

**Comment by wassimj** [comment]
Sorry I cannot help more. I am more comfortable with the Windows installation. And because I am on annual leave, I don’t currently have access to my Linux machine to debug.  My apologies.

**Comment by kq1231** [comment]
No problem! Enjoy your time. I can understand bro. I just want to thank you for replying to me and clearing my doubts.

**Comment by brunopostle** [comment]
Hi, the homemaker add-on should install topologicpy, then all you need to do is download the topologic-sverchok zip and install it as a blender add-on.

It could be my latest homemaker add-on is broken, I'll check

**Comment by brunopostle** [comment]
This works for me on a fresh fedora system: using a downloaded blender binary, then installing the [latest blenderbim snapshot](https://github.com/IfcOpenShell/IfcOpenShell/releases), then installing the [latest homemaker add-on](https://github.com/brunopostle/homemaker-addon/releases) - At this point you can either run the Homemaker add-on or open the python console in blender and type `import topologic` - If this doesn't work there will be messages in the shell where you started blender, please paste them here.

If this is working, you should be able to download and install [sverchok](https://github.com/nortikin/sverchok/archive/refs/heads/master.zip) and the [topologic sverchok add-on](https://github.com/wassimj/topologicsverchok/archive/refs/heads/main.zip)

**Comment by kq1231** [comment]
Okay! Thanks!

**Comment by kq1231** [comment]
I'll try and let you know.

**Comment by kq1231** [comment]
![Screenshot_20210810_123740](https://user-images.githubusercontent.com/88677687/128827456-2d9c2886-1b2a-4f19-87a2-bc5b69e00cec.png)
I have been able to install BlenderBIM.

**Comment by kq1231** [comment]
But when I click on homemaker, this happens: 
![Screenshot_20210810_123921](https://user-images.githubusercontent.com/88677687/128827673-5c92bf6e-a685-4c0b-b408-efab991b1233.png)
![Screenshot_20210810_123957](https://user-images.githubusercontent.com/88677687/128827882-0a1b0961-ef1d-4eec-a07b-e86941d8f067.png)

**Comment by kq1231** [comment]
And neither am I able to install topologic due the error I mentioned at the very top of this page.

**Comment by brunopostle** [comment]
Are you installing the homemaker add-on zip from the release link I posted earlier? This is a 60 MB download that includes topologic etc..

If you just download from the main GitHub code link, you don't get any of this stuff.

**Comment by kq1231** [comment]
Oh! Thanks! I got the problem. So I was installing it the wrong way.

**Comment by kq1231** [comment]
So how do I downloas it then?

**Comment by kq1231** [comment]
Do I have to clone it?

**Comment by kq1231** [comment]
Wait. I got access to the 64.8 mb file. Let me try to install it and I will let you know.

**Comment by kq1231** [comment]
I don't know if you will believe this but now it's giving me a new problem. 
![Screenshot_20210810_191137](https://user-images.githubusercontent.com/88677687/128882336-5955db0e-9c84-48a1-b98d-9a0edc98eab6.png)

**Comment by kq1231** [comment]
Wait! It's installed! Finally.

**Comment by kq1231** [comment]
Okay so I've been able to install homemaker topologise addon. But how do I get the topologic nodes?

**Comment by brunopostle** [comment]
See the links I posted earlier in the thread, you need the 60MB homemaker add-on, which just happens to have all the libraries needed by the sverchok nodes. The Homemaker add-on requires blenderbim to load, you probably don't need blenderbim yourself, but this is how it is for now. The Topologic Sverchok nodes can be downloaded as zipped source code, but installed like any other blender add-on, see my earlier link (you also need sverchok).

It shouldn't matter how you got your blender, but the blenderbim binaries are only available for python 3.7 and 3.9, so this affects everything else. The official blender downloads are 3.7 and 3.9, depending on how old they are.

**Comment by kq1231** [comment]
I have found the problem! The problem is that I have sverchok installed and it works perfectly fine but when I topologic, it gives me an error. The reason why it gives me that error is that the addon is trying to locate topologic's files but since I don't have Blender installed on my system and I'm just running it on its own, topologic cannot find the files as they are not located at the place it is searching for. 
In my case, blender is in my downloads folder but topologic is searching for its files in the system instead of searching it in the downloads folder where blender is actually located. I am extremely new to python and I can't understand the code. If we can change this code, I believe the problem will be solved. 
Actual location: home/kifayat123/Downloads/blender3.0.../scripts/addons/topologicsverchok-main/\\topologicPy\\site-packages
The wrong location topologic is searching for: home/kifayat123/.config/blender/3.0/scripts/addons/topologicsverchok-main\\topologicPy\\site-packages
We need to change the init.py file which is in the topologicsverchok-main.zip. You have to compress it to get access to the init.py file. Can anyone please edit this file ? @wassimj or @brunopostle

**Comment by kq1231** [comment]
I am a complete noob at python otherwise i might have done it myself.

**Comment by kq1231** [comment]
I am a complete noob at python otherwise I might have done it myself.

**Comment by kq1231** [comment]
I am a complete noob at python otherwise i might have done it myself.

**Comment by kq1231** [comment]
I am a complete noob at python otherwise I might have done it myself.

**Comment by wassimj** [comment]
try the following in the blender console. If it works, put that in the init.py file for topologic-sverchok

import sys
sys.append.path("place here the path to the site-packages folder that has the topologic.egg file.. don't include the egg file itself, stop at the enclosing folder")

**Comment by kq1231** [comment]
![Screenshot_20210811_174042](https://user-images.githubusercontent.com/88677687/129030249-f61dfccf-6110-4ccd-8f08-a04e619f284e.png)
This is what's happening.

**Comment by kq1231** [comment]
However, I am able to import topologic by typing: 
import topologic 
That works fine. But what does this indicate

**Comment by kq1231** [comment]
I have been able to install homemaker but topologic nodes don't show up in sverchok.
![Screenshot_20210811_175015](https://user-images.githubusercontent.com/88677687/129031548-4c111d48-8cc8-44de-810b-d5d48c7b35b7.png)

**Comment by wassimj** [comment]
Sorry. that was supposed to be sys.path.append. I flipped them around. But since "import topologic" works.. then really all you should need to do is to go to: https://github.com/wassimj/topologicsverchok and then from the Green Code button, choose download ZIP (Do NOT UNZIP it). Then install that ZIP file in Blender Add-ons as usual (go to preferences, Add-ons, click install, and choose the ZIP file).

**Comment by kq1231** [comment]
If you want, you can even get remote access to my laptop. I have teamviewer installed. I just want topologic because I really want to learn it.

**Comment by wassimj** [comment]
Hi @kq1231  Please book a time to see me next Tuesday using this [Link](http://jabiw.ycb.me) and we can try to sort it out.

**Comment by wassimj** [comment]
Actually, I have time today if you want. I am in Greece so please work out the timezone difference.

**Comment by kq1231** [comment]
Yeah Im okay if you can come today. I'm here in pakistan and it's 12:34 in the morning. What's the time at your place?
